### ISSUE TITLE
Install documentation page mismatch

### ISSUE THREAD

**Issue by villares** [shoebot/shoebot#325]
Sorry if I'm missing something, but the page at
https://shoebot.readthedocs.io/en/latest/install.html
looks different (outdated?) from 
https://github.com/shoebot/shoebot/blob/master/doc/source/install.rst

**Comment by rlafuente** [comment]
Hey, good find! This is a pretty important slight, I'm checking the ReadTheDocs settings to figure this out.

**Comment by rlafuente** [comment]
Trying to build the docs locally, I have an error, which is probably why ReadTheDocs does not have an up to date version...

```
Running Sphinx v3.4.3
loading pickled environment... failed
failed: build environment version not current
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 14 source files that are out of date
updating environment: [new config] 14 added, 0 changed, 0 removed
reading sources... [ 21%] commands                                                                                                                                                                                  
Exception occurred:
  File "/home/rlafuente/.virtualenvs/shoebot3/lib/python3.9/site-packages/sphinxcontrib/shoebot.py", line 109, in run
    print("oops %e" % e)
TypeError: must be real number, not FileNotFoundError
```

Looking into this one now.

**Comment by rlafuente** [comment]
The previous error is not really related to this issue (was using outdated versions), but fixed anyway.

Building the current docs in RTD complains with the following log:

```
no previously-included directories found matching 'docs/_build'
warning: no files found matching 'README' under directory 'tests'
warning: no files found matching 'README' under directory 'examples'
zip_safe flag not set; analyzing archive contents...
cairo.__pycache__.__init__.cpython-37: module references __file__
cairo.__pycache__._cairo.cpython-37: module references __file__
setup options: 
with_pgi:       no


setuptools>=18.8
pycairo>=1.18.1
meta==1.0.2
Pillow>=6.0.0
pubsub==0.1.2
pygobject>=3.32.1

Installed /home/docs/checkouts/readthedocs.org/user_builds/shoebot/checkouts/latest/.eggs/pycairo-1.20.0-py3.7-linux-x86_64.egg
running install
running bdist_egg

[removed redundant shoebot install messages, nothing unusual here]

Installed /home/docs/checkouts/readthedocs.org/user_builds/shoebot/envs/latest/lib/python3.7/site-packages/shoebot-1.3.1-py3.7.egg
Processing dependencies for shoebot==1.3.1
Searching for pygobject>=3.32.1
Reading https://pypi.org/simple/pygobject/
Downloading https://files.pythonhosted.org/packages/3a/a7/de282a4aaedba59d60a895a7821e6497b39cbdfa94a352776ff45ffc6e6f/PyGObject-3.38.0.tar.gz#sha256=051b950f509f2e9f125add96c1493bde987c527f7a0c15a1f7b69d6d1c3cd8e6
Best match: PyGObject 3.38.0
Processing PyGObject-3.38.0.tar.gz
Writing /tmp/easy_install-61gq357h/PyGObject-3.38.0/setup.cfg
Running PyGObject-3.38.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-61gq357h/PyGObject-3.38.0/egg-dist-tmp-pboz9vm0
Package gobject-introspection-1.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gobject-introspection-1.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gobject-introspection-1.0' found
error: Setup script exited with Command '('pkg-config', '--print-errors', '--exists', 'gobject-introspection-1.0 >= 1.46.0')' returned non-zero exit status 1.

Try installing it with: 'sudo apt install libgirepository1.0-dev'
```

I think @stuaxo will know what to do here. It's been a long time since I hacked on the docs, but the problem seems to be a missing package dependency. From [this ticket](https://github.com/readthedocs/readthedocs-docker-images/issues/114) it seems that we could call apt directly from sphinx's `conf.py`, but it's a hack. Not sure if there's a simpler way to install the missing `libgirepository1.0-dev`

**Comment by speters33w** [comment]
@stuaxo and I are working with Shoebot to work on Windows, the newest github has some changes from readthedocs.io, the newer revision isn't ready for readthedocs.io yet.
There's work being done, especially on the gedit extension.

Try this for the local build:

From the docs directory:

`pip install -r requirements.txt`

Then build docs:
`sphinx-build -b html -d build/doctrees source build/html`

There's another issue recently opened about readthedocs.io as well, [#320](https://github.com/shoebot/shoebot/issues/320)

**Comment by stuaxo** [comment]
@rlafuente can you remember if it's possible to install Debian packages under RTD ?

**Comment by rlafuente** [comment]
> @stuaxo and I are working with Shoebot to work on Windows, the newest github has some changes from readthedocs.io, the newer revision isn't ready for readthedocs.io yet.

I *think* this has to do with RTD not having a necessary dependency -- see my error dump above, which I got from the RTD web build interface. I can build locally without issue.

> can you remember if it's possible to install Debian packages under RTD ?

I couldn't find a way. According to [this issue](https://github.com/readthedocs/readthedocs-docker-images/issues/114) it's not supported, but in the thread there is the mention of a hack with `conf.py` to make it work, but I couldn't figure out how. This [other issue](https://github.com/readthedocs/readthedocs.org/issues/6919) kind of makes it clear that it's not supported. Not sure where we can go from here -- how vital is `libgirepository`?

**Comment by stuaxo** [comment]
That's an interesting rabbit hole :) I'm starting to remember when we set this up years ago now..

Maybe a better way to frame this is, where can I see what packages are installed on RTD?

When we set this up I think it had most of the packages we needed but they were out of date, but that was more probably more than 5 years ago, so their packages will be more up to date and we moved to gtk3 and python 3 in the meantime.

I'd need to check, but I think we might work with just pycairo (which we can install from pip).
If we want to render text we want Pango too, from apt get.

**Comment by rlafuente** [comment]
I think [this dockerfile](https://github.com/readthedocs/readthedocs-docker-images/blob/master/Dockerfile) is what's behind RTD's build system, requirements are on line 15

**Comment by stuaxo** [comment]
It looks like the `conf.py` way of installing extra packages on readthedocs unsupported.
Other projects succeeded in getting them to add things, so we should try this avenue.

I had a poke at docker, to find a minimum list of extra packages we need.
We use `pycairo` and `pygobject` so we need their dependencies.   We also use `pango` and `rsvg` via gobject bindings.

We should ask if they can add these packages so we can use `sphinxcontrib-shoebot`.
```
libgirepository1.0-dev   # This lets us install pygobject and pycairo, both of which only compile from source.
gir1.2-pango-1.0         # Gobject bindings for Pango, which we use to render text
gir1.2-rsvg-2.0          # Gobject bindings for Rsvg, so we can output to svg
```

Are you up for seeing asking if they will add the packages, I don't think I'm going to have the time / bandwidth for a bit ?

**Comment by stuaxo** [comment]
Added a ticket here
https://github.com/readthedocs/readthedocs-docker-images/issues/162

**Comment by rlafuente** [comment]
#336 is a way out of this.

**Comment by rlafuente** [comment]
Fixed #336, so this is not a thing anymore
